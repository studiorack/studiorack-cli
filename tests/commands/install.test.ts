import { expect, test } from 'vitest';
import { cli } from '../shared';
import { RegistryType } from '@open-audio-stack/core';

test('Install package', async () => {
  expect(cli(RegistryType.Plugins, 'install', 'surge-synthesizer/surge')).toMatchSnapshot();
  expect(cli(RegistryType.Plugins, 'install', 'surge-synthesizer/surge@1.3.1')).toMatchSnapshot();
  expect(cli(RegistryType.Plugins, 'install', 'surge-synthesizer/surge@0.0.0')).toMatchSnapshot();
});
