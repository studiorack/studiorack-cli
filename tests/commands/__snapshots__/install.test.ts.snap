// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Install package 1`] = `
┌────┬──────────┬─────────┬────────────┬─────────┬───────────────────────────────┐
│ Id │ Name     │ Version │ Date       │ License │ Tags                          │
├────┼──────────┼─────────┼────────────┼─────────┼───────────────────────────────┤
│ -  │ Surge XT │ 1.3.4   │ 2024-08-11 │ gpl-3.0 │ Instrument, Synth, Modulation │
└────┴──────────┴─────────┴────────────┴─────────┴───────────────────────────────┘
`;

exports[`Install package 2`] = `
┌────┬──────────┬─────────┬────────────┬─────────┬───────────────────────────────┐
│ Id │ Name     │ Version │ Date       │ License │ Tags                          │
├────┼──────────┼─────────┼────────────┼─────────┼───────────────────────────────┤
│ -  │ Surge XT │ 1.3.1   │ 2024-03-02 │ gpl-3.0 │ Instrument, Synth, Modulation │
└────┴──────────┴─────────┴────────────┴─────────┴───────────────────────────────┘
`;

exports[`Install package 3`] = `No result found`;
